### PR TITLE
Implement the actions required workflows APIs

### PR DESCRIPTION
**Pull request by avivek**
Have implemented the API's that would enable managing Required Workflows.

Merging this Fixes google/go-github#2729

### CODE REVIEW

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:27]: Please add godoc-style comments to all exported fields.

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:146]: On these extremely long method names, let's please shorten "Repositories" to "Repos".

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @avivek !
It looks like the linter also caught a bunch of things (let me know if you can't see them), so I'm going to cut my review short and let you address them.  Please let me know when you are ready for a full review.

File: github/actions_required_workflows.go
Comment (avivek) [reply github/actions_required_workflows.go]: Yes I have now renamed the method names.

File: github/actions_required_workflows.go
Comment (avivek) [reply github/actions_required_workflows.go:28]: Done.

File: github/actions_required_workflows.go
Comment (avivek) [reply github/actions_required_workflows.go]: Done.

File: github/actions_required_workflows.go
Comment (avivek) [reply github/actions_required_workflows.go]: Done.

File: github/actions_required_workflows_test.go
Comment (avivek) [reply github/actions_required_workflows_test.go:6]: Added.

File: github/actions_required_workflows.go
Comment (avivek) [reply github/actions_required_workflows.go]: Done

**Comment by avivek** [comment]
Hi @gmlewis ,
I have made the changes and requested re-review. 
Pls let me know if there is any further changes required.
Thanks.

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:35]:
```suggestion
	WorkflowFilePath      *string         `json:"workflow_file_path,omitempty"`
```

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:38]:
```suggestion
	SelectedRepositoryIDs *SelectedRepoIDs `json:"selected_repository_ids,omitempty"`
```

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:41]:
```suggestion
// RequiredWorkflowSelectedRepos represents the repos that a required workflow is applied to.
```

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:202]:
```suggestion
func (s *ActionsService) RemoveRepoFromRequiredWorkflow(ctx context.Context, org string, requiredWorkflowID, repoID int64) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @avivek !
Just a few more tweaks, please.

**Comment by gmlewis** [comment]
It looks like you might also have to run `gofmt` on the files you modified and run `go generate ./...` and push the results (not force-push) to the repo to get the linter to pass its tests.  Thanks.
See CONTRIBUTING.md for more details.

**Comment by avivek** [comment]
Hi @gmlewis ,
Thanks for the quick review, the changes have now been committed.
I have requested for you re-review.

File: github/actions_required_workflows.go
Comment (gmlewis) [inline github/actions_required_workflows.go:190]: Whups, looks like I missed one.

```suggestion
func (s *ActionsService) AddRepoToRequiredWorkflow(ctx context.Context, org string, requiredWorkflowI, repoID int64) (*Response, error) {
```

Comment (avivek) [reply github/actions_required_workflows.go]: Done.

**Comment by avivek** [comment]
Hi @gmlewis ,
Addressed your comment. Hope all set now.
Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @avivek !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
